docker/README.md

```diff
@@ -79,3 +79,26 @@ Try making a new notebook by clicking the new notebook button and typing "my fir
 Now, if you open up a new terminal window and run `ls -l ~/polynote-docker/notebooks`, you should see `my first notebook.ipynb` on your host machine!
 
 
+# Building the Image
+
+To build the image from the Dockerfiles included here, run the following series of commands from the `docker` directory:
+
+We tag these images with the name `polynote-local`.
+
+## Base image
```

Comment:
Might want to mention the dev dockerfile https://github.com/polynote/polynote/blob/master/docker/dev/Dockerfile  as well. It has some instructions inside that I guess we could pull out into the readme if that's preferred.

Reply:
I actually had a hard time understanding those instructions. How is the `dev` one built?

Reply:
Sorry 😞 Let me see whether I can do a better job explaining this too. 

The `dev` docker image is based on a locally-created Polynote distribution (tarball). So, in order to build the `dev` docker image, you need to first create the Polynote distribution using `sbt dist`. 

After that command completed, the distribution tarball lives in `target/scala-2.11`. Since `docker build` will send the entire current directory as build context to the daemon, it's best not to run it in the root directory of the project. So, we want to `cd` over to that directory. 

Finally we want to run the command. So, we run `docker build` and point to the `dev` Dockerfile. 

Putting it all together:
```
git clone https://github.com/polynote/polynote.git
cd polynote/
sbt dist
cd target/scala-2.11/
docker build -t polynote:dev -f ../../docker/dev/Dockerfile .
```

Reply:
that's somewhat along the lines of what I was guessing from the comments, but it helps so much to see it fleshed out. The confusion stems from total lack of familiarity with what `sbt` is/does. And also maybe the fact that I haven't worked with Scala or Spark, so I've never had to build things outside of the Dockerfile before running `docker build`, and I imagine I'm not alone in that respect (perhaps it helps to clarify that my background is in mathematics, I've mostly done ml research but now am in consulting, picked up dev ops literally as a hobby but it's proven useful at the office). 

Why does it need to be built outside of docker for `dev` but not `base`?

Reply:
I wouldn't worry about changing `dev` - since it's never published and meant for local usage it's probably better to make it simple. 

[`sbt`](https://www.scala-sbt.org/) is a build tool popular among Scala developers. It's used to build [jars](https://en.wikipedia.org/wiki/JAR_(file_format)). 

`docker build` is used to create Docker containers, so it's unrelated to the applications you put inside the container. (Of course, you could run `sbt` inside your `Dockerfile` if you wanted to, since you can run anything inside your `Dockerfile`).

> Why does it need to be built outside of docker for dev but not base?

Every time we make a release we have GitHub CI build the Polynote tarballs. This process uses `sbt`. When `base` is built, it just downloads those tarballs, so there's no need for you to run `sbt` yourself. 

However, `dev` is meant to be used while developing on Polynote, so it can't use those pre-built binaries. You need to build your own binaries (in order for the changes you've been making locally to be reflected). Thus, you need to use `sbt dist` to build a local version of the binaries, which then get copied into the `dev` docker image that you use locally for testing your changes.